packages/google_maps_flutter/google_maps_flutter_android/android/src/test/java/io/flutter/plugins/googlemaps/GroundOverlaysControllerTest.java

```diff
@@ -0,0 +1,150 @@
+    byte[] bmpData = Base64.decode(base64Image, Base64.DEFAULT);
+
+    return new Messages.PlatformGroundOverlay.Builder()
+        .setImage(
+            new Messages.PlatformBitmap.Builder()
+                .setBitmap(
+                    new Messages.PlatformBitmapBytesMap.Builder()
+                        .setBitmapScaling(Messages.PlatformMapBitmapScaling.AUTO)
+                        .setImagePixelRatio(2.0)
+                        .setByteData(bmpData)
+                        .setWidth(100.0)
+                        .build())
+                .build())
+        .setBearing(1.0)
+        .setZIndex(1L)
+        .setVisible(true)
+        .setTransparency(1.0)
+        .setClickable(true);
+  }
+
+  @Before
+  public void setUp() {
+    mockCloseable = MockitoAnnotations.openMocks(this);
+    Context context = ApplicationProvider.getApplicationContext();
+    AssetManager assetManager = context.getAssets();
+    Messages.MapsCallbackApi flutterApi =
+        spy(new Messages.MapsCallbackApi(mock(BinaryMessenger.class)));
+    controller =
+        spy(
+            new GroundOverlaysController(
+                flutterApi, assetManager, 1.0f, bitmapDescriptorFactoryWrapper));
+    googleMap = mock(GoogleMap.class);
+    controller.setGoogleMap(googleMap);
+    when(bitmapDescriptorFactoryWrapper.fromBitmap(any())).thenReturn(mockBitmapDescriptor);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    mockCloseable.close();
+  }
+
+  @Test
+  public void controller_AddChangeAndRemoveGroundOverlay() {
+    final GroundOverlay groundOverlay = mock(GroundOverlay.class);
+    final String googleGroundOverlayId = "abc123";
+    final float transparency = 0.1f;
+
+    when(groundOverlay.getId()).thenReturn(googleGroundOverlayId);
+    when(googleMap.addGroundOverlay(any(GroundOverlayOptions.class))).thenReturn(groundOverlay);
+
+    controller.addGroundOverlays(
+        Collections.singletonList(
+            defaultGroundOverlayBuilder()
+                .setGroundOverlayId(googleGroundOverlayId)
+                .setTransparency((double) transparency)
+                .build()));
+    Mockito.verify(googleMap, times(1))
+        .addGroundOverlay(Mockito.argThat(argument -> argument.getTransparency() == transparency));
+
+    final float newTransparency = 0.2f;
+    controller.changeGroundOverlays(
+        Collections.singletonList(
+            defaultGroundOverlayBuilder()
+                .setGroundOverlayId(googleGroundOverlayId)
+                .setTransparency((double) newTransparency)
+                .build()));
+    Mockito.verify(groundOverlay, times(1)).setTransparency(newTransparency);
+
+    controller.removeGroundOverlays(Collections.singletonList(googleGroundOverlayId));
+
+    Mockito.verify(groundOverlay, times(1)).remove();
+  }
+
+  // Helper method to generate 1x1 pixel base64 encoded png test image
+  private String generateBase64Image() {
```

Comment:
This can be pulled out of the other file into a test helper file, rather than duplicating the code.